### PR TITLE
reuse: 0.7.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by ethulhu**
###### Motivation for this change

reuse updated to 0.10.1.

###### Things done

- Bumped version.
- Updated SHA256.
- Added new dependency, `setuputils_scm`.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by FlorianFranzen — APPROVED** [review]
Looks good. Builds and runs well. Thank you for your contribution. LGTM!

File: pkgs/tools/package-management/reuse/default.nix [hunk 1]
```diff
@@ -21,6 +21,7 @@ buildPythonApplication rec {
     license-expression
     requests
     setuptools
+    setuptools_scm
```
Comment (geistesk) [inline pkgs/tools/package-management/reuse/default.nix:24]: Perhaps all the Python libraries should be included on their own and not the whole `python3Packages` in the first line. This was already the case before, but could perhaps be changed.

File: pkgs/tools/package-management/reuse/default.nix [hunk 2]
```diff
@@ -4,13 +4,13 @@ with python3Packages;
 
 buildPythonApplication rec {
   pname = "reuse";
-  version = "0.7.0";
+  version = "0.10.1";
```
Comment (geistesk) [inline pkgs/tools/package-management/reuse/default.nix:7]: A newer version was released in the meantime.

**Review by oxzi — COMMENTED** [review]
Perhaps you should update this PR to the latest upstream version. Otherwise, lgtm.

File: pkgs/tools/package-management/reuse/default.nix (hunk 1 above)
Comment (ethulhu) [reply pkgs/tools/package-management/reuse/default.nix:24]: This PR is just the version bump; I would prefer to keep it single-purpose.

**Review by oxzi — APPROVED** [review]
LGTM

File: pkgs/tools/package-management/reuse/default.nix (hunk 1 above)
Comment (geistesk) [reply pkgs/tools/package-management/reuse/default.nix:24]: Okay, but perhaps someone should have another look at this later.

**Comment by ethulhu** [comment]
Gentle ping?

**Comment by Luflosi** [comment]
#89752, a duplicate PR, was merged, so this can probably be closed now.

**Pull request closed without merging**